mne/preprocessing/ica.py

```diff
@@ -601,6 +595,13 @@ def fit(self, inst, picks=None, start=None, stop=None, decim=None,
         self : instance of ICA
             Returns the modified instance.
         """
+        if self.method == 'fastica' and not check_version('sklearn'):
+            raise ImportError(
+                'The scikit-learn package is required for method="fastica".')
+        if self.method == 'picard' and not check_version('picard'):
+            raise ImportError(
+                'The python-picard package is required for method="picard".')
+
```

Comment:
Might as well modernize since we're doing code churn anyway
```suggestion
        req_map = dict(fastica='sklearn', picard='picard')
        for method, lib in req_map.items():
            if self.method == method:
                _require_version(mod, f'use method={repr(method)}')
```

Reply:
OK! You meant to write `lib` instead of `mod`, is there a way I can edit your suggestion before committing?

Reply:
You might as well copy-paste since you also need to add a `from ..utils import _require_version` for my fix to work anyway

Reply:
OK, will do!